### PR TITLE
Remade homepage based on new design

### LINKED ISSUE #10: Add website index
desktop_home_v2_-_hash_labs_v5_-_zeplin

_check design at zeplin to look for mobile layout_

### PR DESCRIPTION
**Pull request by TheMightyPenguin**
# Context
Homepage had to be remade using new design, so this commit issues that by:

- Added lead section.
- Added brands section.
- Added what we do section.
- Added techonologies/tools section.
- Added how section.
- Added CAT (call to action) section.
- Added FAW section.
- Added Jekyll Autoprefixer Gem. It automatically adds vendor prefixes to certain rules, to improve browser compatibility.
- Added Jquery Widont plugin, it automatically adds a non breaking space between the 2 last words of text nodes to prevent orphan words.

All these sections have their own styles, and were done as separate components, so they are easy to modify, remove or reorder. 

Some listing sections were made with flexbox, so some polyfill should be added for it, already created an issue (#21) for that. Also video should be updated once new one with loop is available (#22).

This PR closes #10 

## Links
[Jquery widont](https://davecardwell.co.uk/javascript/jquery/plugins/jquery-widont/)
[Jekyll Autoprefixer Gem](https://github.com/vwochnik/jekyll-autoprefixer)
[Desktop design](https://app.zeplin.io/project.html#pid=5888b73ba8de3a354ef67567&sid=588f6811cb4fd20443ab9447)
[Mobile design](https://app.zeplin.io/project.html#pid=5888b73ba8de3a354ef67567&sid=58b86d91c50cf643a18ef7b2)

## Media
homepage

### CODE REVIEW

**Comment by orlando** [comment]
@TheMightyPenguin can you post a screenshot please?

**Comment by duranmla** [comment]
@TheMightyPenguin in this case I will feel even more comfortable to approve if you share me a ngrok link with your localhost just to be able to play again with the site. (keep the gif there as reference too)

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,4 @@ source 'https://rubygems.org'
 
 gem 'jekyll'
 gem 'rack-jekyll'
+gem 'jekyll-autoprefixer'
```
Comment (duranmla) [inline Gemfile:5]: Everytime we add things like this one I want to see a link within the PR context and an explanation of why we needed so we all will be clear of why we have introduced a new dependency.

File: _includes/footer-scripts.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
 <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.3/jQuery.headroom.min.js"></script>
 <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/wow/1.1.2/wow.min.js"></script>
 <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery.lazyloadxt/1.1.0/jquery.lazyloadxt.min.js"></script>
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/flexibility/2.0.1/flexibility.js"></script>
```
Comment (duranmla) [inline _includes/footer-scripts.html]: Same as before, why we need this one?

File: _includes/section-cat.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="container-fluid section-cat-container">
```
Comment (duranmla) [inline _includes/section-cat.html]: maybe we want to change to CTA instead CAT right? `Call To Action` so let's update the class names that contains "cat"

File: _sass/_components--brands.scss [hunk 4]
```diff
@@ -0,0 +1,59 @@
+$size--brand-quantity:                5;
+$size--brand-width:                   100% / $size--brand-quantity - 1;
```
Comment (duranmla) [inline _sass/_components--brands.scss]: let's use parenthesis on this type of operations like: `100% / ($size--brand-quantity - 1)`

File: _sass/_components--brands.scss [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+$size--brand-width-tablet:            100% / 2 - 1;
```
Comment (duranmla) [inline _sass/_components--brands.scss]: what is `2` and `1` and why `(2 - 1)` I will rather prefer variables for those than the percentages for the images width.

File: _sass/_components--brands.scss [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+$size--brand-width-small:             100%;
```
Comment (duranmla) [inline _sass/_components--brands.scss]: brand-image? or what is the difference between brand and brand-image? there is more variables below that say `brand-image`

File: _sass/_components--footer.scss [hunk 7]
```diff
@@ -61,3 +64,23 @@ $size--footer-padding-top: 60px;
 .footer-link {
   padding: 0;
 }
+
+// Basically applying same padding left to footer logo
+// than the one being applied to header logo, only
+// showing this tablets and below.
```
Comment (duranmla) [inline _sass/_components--footer.scss]: "...in order to keep the aligned on mobile devices" and yes awesome comment

File: _sass/_components--footer.scss [hunk 8]
```diff
@@ -61,3 +64,23 @@ $size--footer-padding-top: 60px;
...
+.footer-logo-small {
+  display: block;
+  position: absolute;
+  left: $space-triple;
+
+  img {
+    // This is the logo image size to match the
+    // header logo.
+    width: 44px;
```
Comment (duranmla) [inline _sass/_components--footer.scss]: so this should be a variable on the variables file don't you think? like "size--brand-logo"

File: _sass/_components--technologies.scss [hunk 9]
```diff
@@ -0,0 +1,57 @@
+$size--number-of-techs:      6;
+$size--tech-width-large:     100% / 6 - 1; // all techs on a single row
```
Comment (duranmla) [inline _sass/_components--technologies.scss]: that comment is great but why we use `-1`? I want to minimize my effort to understand the code while I am reading, we need to make it as clear and obvious as possible

Comment (duranmla) [reply _sass/_components--technologies.scss]: Add this type of comment on the feedback above that has the same approach

File: _sass/_settings--fonts.scss [hunk 10]
```diff
@@ -11,7 +11,7 @@
  // font-import
  // -----------------------------------------------------------------------------
 
-  @import url(https://fonts.googleapis.com/css?family=Montserrat:400,700);
+  @import url('https://fonts.googleapis.com/css?family=Montserrat:300,400,500,700');
```
Comment (duranmla) [inline _sass/_settings--fonts.scss]: damn do we actually need all of that?

**Review by duranmla — CHANGES_REQUESTED** [review]
The bottom line is to keep in mind to make the code clear, remember that you are writing code for humans, always aim to make your code as readable as possible.

But overall good work mate, today is the last day for this, so let's make it happen.

**Review by duranmla — APPROVED** [review]

**Comment by duranmla** [comment]
@TheMightyPenguin LGTM

**Pull request merged**